### PR TITLE
Demo script for TTS

### PR DESCRIPTION
**Pull request by takenori-y**
This is a TTS version of #581.

### CODE REVIEW

**Comment by kan-bayashi** [comment]
Awesome

File: utils/synth_wav.sh [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+train_cmd=
+decode_cmd=
+
+. ./cmd.sh
+
+txt=$1
+download_dir=${decode_dir}/download
+
+if [ $# -gt 1 ]; then
+    echo "Usage: $0 <wav>"
```
Comment (kan-bayashi) [inline utils/synth_wav.sh]: <text> ?

File: utils/synth_wav.sh [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+
+if [ ${stage} -le 3 ] && [ ${stop_stage} -ge 3 ] && [ ${use_input_wav} ]; then
+    echo "stage 3: x-vector extraction"
+
+    utils/copy_data_dir.sh ${decode_dir}/data ${decode_dir}/data2
+    echo "$base ${input_wav}" > ${decode_dir}/data2/wav.scp
+    utils/data/resample_data_dir.sh 16000 ${decode_dir}/data2
+    steps/make_mfcc.sh \
+        --write-utt2num-frames true \
+        --mfcc-config conf/mfcc.conf \
```
Comment (kan-bayashi) [inline utils/synth_wav.sh]: Maybe `mfcc.conf` is missing?

Comment (kan-bayashi) [reply utils/synth_wav.sh]: OK, I will change the directory.

File: utils/synth_wav.sh (hunk 1 above)
Comment (kan-bayashi) [reply utils/synth_wav.sh]: Maybe `-ne` is better?

File: utils/synth_wav.sh [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+# Check file existence
+if [ ! -f "${cmvn}" ]; then
+    echo "No such CMVN file: ${cmvn}"
+    exit 1
+fi
+if [ ! -f "${dict}" ]; then
+    echo "No such dictionary: ${dict}"
+    exit 1
+fi
+if [ ! -f "${input_wav}" ] && [ ${use_input_wav} ]; then
```
Comment (kan-bayashi) [inline utils/synth_wav.sh]:
```suggestion
if [ ! -f "${input_wav}" ] && ${use_input_wav}; then
```

File: utils/synth_wav.sh [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+
+if [ ${stage} -le 2 ] && [ ${stop_stage} -ge 2 ]; then
+    echo "stage 2: Json Data Preparation"
+
+    feat_synth_dir=${decode_dir}/dump
+    data2json.sh --feat ${feat_synth_dir}/feats.scp \
+        ${decode_dir}/data ${dict} > ${feat_synth_dir}/data.json
+fi
+
+if [ ${stage} -le 3 ] && [ ${stop_stage} -ge 3 ] && [ ${use_input_wav} ]; then
```
Comment (kan-bayashi) [inline utils/synth_wav.sh]:
```suggestion
if [ ${stage} -le 3 ] && [ ${stop_stage} -ge 3 ] && ${use_input_wav}; then
```

File: utils/synth_wav.sh [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+    echo "Usage: $0 <wav>"
+    exit 1;
+fi
+
+set -e
+set -u
+set -o pipefail
+
+case "${models}" in
+    "libritts.v1")
```
Comment (kan-bayashi) [inline utils/synth_wav.sh]: Even when I want to use my own model, `use_input_wav` is always set as `true`.

File: utils/synth_wav.sh (hunk 4 above)
Comment (kan-bayashi) [reply utils/synth_wav.sh]: `if [ ${stage} -le 3 ] && [ ${stop_stage} -ge 3 ] && [ ${use_input_wav} ]` is not worked.
`[ ${use_input_wav} ]` will be true even if `use_input_wav=false`.

File: utils/synth_wav.sh (hunk 3 above)
Comment (kan-bayashi) [inline utils/synth_wav.sh]:
```suggestion
if [ ! -f "${input_wav}" ] && ${use_input_wav}; then
```

File: utils/synth_wav.sh [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+    mkdir -p ${decode_dir}/data
+    echo "$base ${decode_dir}/data/dummy.wav" > ${decode_dir}/data/wav.scp
+    echo "X $base" > ${decode_dir}/data/spk2utt
+    echo "$base X" > ${decode_dir}/data/utt2spk
+    echo -n "$base " > ${decode_dir}/data/text
+    cat $txt >> ${decode_dir}/data/text
+fi
+
+if [ ${stage} -le 1 ] && [ ${stop_stage} -ge 1 ]; then
+    echo "stage 1: Dummy Feature Generation"
```
Comment (kan-bayashi) [inline utils/synth_wav.sh]: I want to remove this part.
Why don't you add following lines in just before this part?
https://github.com/espnet/espnet/blob/c69b202e9cbdd0904c35ad256a44dcc177a5a318/utils/data2json.sh#L103
```bash
# modified as follows:
    if [ -z "$(find "${tmpdir}/${intype}" -name "*.scp")" ]; then
        continue
    fi
    if [ ${intype} != other ]; then 
```

Then, we can get  json like this:
```json
/home/t_hayashi/workspace5/work/espnet_dev/utils/data2json.sh decode/sample1/data decode/download/libritts.v1/data/lang_1char/train_clean_460_units.txt
{
    "utts": {
        "sample1": {
            "input": [],
            "output": [
                {
                    "name": "target1",
                    "shape": [
                        105,
                        78
                    ],
                    "text": "This example text is a how-to-write sample, so please add to and delete from it as required before using.",
                    "token": "T h i s <space> e x a m p l e <space> t e x t <space> i s <space> a <space> h o w - t o - w r i t e <space> s a m p l e , <space> s o <space> p l e a s e <space> a d d <space> t o <space> a n d <space> d e l e t e <space> f r o m <space> i t <space> a s <space> r e q u i r e d <space> b e f o r e <space> u s i n g .",
                    "tokenid": "34 50 51 61 13 47 66 43 55 58 54 47 13 62 47 66 62 13 51 61 13 43 13 50 57 65 8 62 57 8 65 60 51 62 47 13 61 43 55 58 54 47 7 13 61 57 13 58 54 47 43 61 47 13 43 46 46 13 62 57 13 43 56 46 13 46 47 54 47 62 47 13 48 60 57 55 13 51 62 13 43 61 13 60 47 59 63 51 60 47 46 13 44 47 48 57 60 47 13 63 61 51 56 49 9"
                }
            ],
            "utt2spk": "X"
        }
    }
}

```

File: utils/synth_wav.sh (hunk 4 above)
Comment (takenori-y) [reply utils/synth_wav.sh]: I see. I will fix it.

File: utils/synth_wav.sh (hunk 5 above)
Comment (takenori-y) [reply utils/synth_wav.sh]: To solve the problem, I set the initial value of `models` to empty.

File: utils/synth_wav.sh (hunk 3 above)
Comment (takenori-y) [reply utils/synth_wav.sh]: Fixed

File: utils/synth_wav.sh (hunk 2 above)
Comment (takenori-y) [reply utils/synth_wav.sh]: What do you think about that we include `mfcc.conf` and `vad.conf` in a tar.gz file?

File: utils/synth_wav.sh (hunk 1 above)
Comment (takenori-y) [reply utils/synth_wav.sh]: Fixed

File: utils/synth_wav.sh (hunk 6 above)
Comment (takenori-y) [reply utils/synth_wav.sh]: Thank you for your suggestion. It will make the script simple.

**Comment by kan-bayashi** [comment]
I will also update my pretained models by including dict.

File: utils/synth_wav.sh (hunk 5 above)
Comment (takenori-y) [reply utils/synth_wav.sh]: Fixed

**Comment by takenori-y** [comment]
Thank you for your review. I have updated the script. An example wav file has been included in the model package.

File: utils/synth_wav.sh [hunk 7]
```diff
@@ -0,0 +1,225 @@
+#!/bin/bash
+
+# Copyright 2019 Nagoya University (Takenori Yoshimura)
+#  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
+
+if [ ! -f path.sh ] || [ ! -f cmd.sh ]; then
+    echo "Please change directory to e.g., egs/ljspeech/tts1"
```
Comment (kan-bayashi) [inline utils/synth_wav.sh]:
```suggestion
    echo "Please change directory to e.g., egs/libritts/tts1"
```

File: utils/synth_wav.sh [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+decode_cmd=
+
+. ./cmd.sh
+
+txt=$1
+download_dir=${decode_dir}/download
+
+if [ $# -ne 1 ]; then
+    echo "Usage: $0 <text>"
+    exit 1;
```
Comment (kan-bayashi) [inline utils/synth_wav.sh]: Why don't you add example usage?
```bash
echo "Example:"
echo "    echo \"This is the demo script of ESPnet end-to-end text-to-speech model.\" > example.txt"
echo "    $0 exmaple.txt"
```

File: utils/synth_wav.sh (hunk 2 above)
Comment (kan-bayashi) [reply utils/synth_wav.sh]: I think current style is acceptable.

Comment (takenori-y) [reply utils/synth_wav.sh]: OK.

File: utils/synth_wav.sh (hunk 7 above)
Comment (takenori-y) [reply utils/synth_wav.sh]: Fixed

File: utils/synth_wav.sh (hunk 8 above)
Comment (takenori-y) [reply utils/synth_wav.sh]: Added

**Comment by kan-bayashi** [comment]
I will merge.

**Pull request merged**